### DISCUSSION TITLE
Speeding up static type checking CI action

### DISCUSSION THREAD

**Discussion by erictraut** [colour-science/colour#1301] [General]
I noticed that this repo has a github action that uses pyright for static type checking. I'm the primary maintainer of pyright, and I recently added multi-core support. You can enable it using the `--threads` command-line option. Depending on how many cores your CI servers have, this could speed up type checking by 4x or more. For example, on my local machine, static type checking for this repo drops from 44s to 12s with this option.

**Comment by thomasmansencal** [comment]
Fantastic idea @erictraut! I will roll that across the board.

**Reply by thomasmansencal** [reply]
2m16.217s

vs

1m12.564s

on my machine!

**Reply by erictraut** [reply]
Incidentally, there was a recent change in the `colour-science` repo that defeated some of my previous performance optimizations in pyright. You may have noticed that type checking times [increase significantly](https://github.com/microsoft/pyright/issues/9109). I've improved the performance optimizations in pyright, so you should see those times come down again.

Also, the `colour-science` repo was recently added to the [mypy-primer](https://github.com/hauntsaninja/mypy_primer/blob/17dd6bc9df67390122dfd629550698d7ccc4f8e8/mypy_primer/projects.py#L1227) test tool. Mypy-primer runs as a CI action for pyright, mypy, and typeshed. Now that `colour-science` is part of that test suite, any future regressions in type checking performance should be caught more quickly.

**Reply by thomasmansencal** [reply]
This is quite exciting, I'll let the team know!
